test/testnullpointer.cpp

```diff
@@ -2503,7 +2503,7 @@ class TestNullPointer : public TestFixture {
 
     void ticket6505() {
         check("void foo(MythSocket *socket) {\n"
-              "  bool do_write=false;\n"
+              "  bool do_write=0;\n"
```

Comment:
With "false" the function valueFlowSetConstantValue does not set the value.
For a real run the "false" is probably first simplified to "0". However, in test runs the simplification is not executed before the code is checked.